tests/mechanic/java_resolver_test.py

```diff
@@ -53,9 +57,22 @@ def test_resolves_java_home_for_bundled_jdk(self):
         cfg = config.Config()
         cfg.add(config.Scope.application, "mechanic", "runtime.jdk", "bundled")
 
-        major, java_home = java_resolver.java_home("12,11,10,9,8", cfg)
+        major, java_home = java_resolver.java_home("12,11,10,9,8",
+                                                   specified_runtime_jdk=cfg.opts("mechanic", "runtime.jdk"),
+                                                   provides_bundled_jdk=True)
 
         # assumes most recent JDK
         self.assertEqual(major, 12)
         # does not set JAVA_HOME for the bundled JDK
         self.assertEqual(java_home, None)
+
+    def test_disallowed_bundled_jdk(self):
+
+        cfg = config.Config()
+        cfg.add(config.Scope.application, "mechanic", "runtime.jdk", "bundled")
+        cfg.add(config.Scope.application, "mechanic", "car.names", ["default"])
+
+        with self.assertRaises(exceptions.SystemSetupError) as ctx:
+            java_resolver.java_home("12,11,10,9,8", specified_runtime_jdk=cfg.opts("mechanic", "runtime.jdk"))
```

Comment:
There is no need for the `Config` object anymore and therefore we should simplify the test and only pass the actual value here.